src/Handler/QueryMonitorHandler.php

```diff
@@ -0,0 +1,55 @@
+<?php # -*- coding: utf-8 -*-
+/*
+ * This file is part of the Wonolog package.
+ *
+ * (c) Inpsyde GmbH
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Inpsyde\Wonolog\Handler;
+
+use Monolog\Handler\AbstractProcessingHandler;
+use Psr\Log\LogLevel;
+
+/**
+ * Class QueryMonitorHandler.
+ *
+ * @package BoxUk\Plugins\Base
+ */
+class QueryMonitorHandler extends AbstractProcessingHandler {
+	private static $level_map = array(
+		LogLevel::DEBUG => 'qm/debug',
+		LogLevel::INFO => 'qm/info',
+		LogLevel::NOTICE => 'qm/notice',
+		LogLevel::WARNING => 'qm/warning',
+		LogLevel::ERROR => 'qm/error',
+		LogLevel::CRITICAL => 'qm/critical',
+		LogLevel::ALERT => 'qm/alert',
+		LogLevel::EMERGENCY => 'qm/emergency',
+	);
+
+	/**
+	 * Write the log to the Query Monitor log.
+	 *
+	 * @param array $record The record to write.
+	 *
+	 * @return void
+	 */
+	protected function write( array $record ) {
+		$level = $this->map_level( $record['level_name'] );
+		do_action( $level, $record['message'], $record['context'] );
```

Comment:
We can't be 100% sure 'level_name', 'message' and 'context' keys are available in the `$record` array.

So we either need defaults or check for presence. Something like:

```php
if ( empty($record['message']) || ! is_string($record['message']) ) {
    return;
}
if ( !isset($record['level_name']) || ! is_string($record['level_name']) ) {
    $record['level_name'] = LogLevel::DEBUG;
}
if ( !isset($record['context']) || ! is_array($record['context']) ) {
    $record['context'] = [];
}

$qmLevel = $this->map_level( $record['level_name'] );
do_action( $qmLevel, $record['message'], $record['context'] );
``` 

We need to be extra careful considering that there's probably nothing that can log warnings from the logger code.

Reply:
Have done and added a few more tests.